packages/block-library/src/navigation-link/index.php

```diff
@@ -392,6 +393,27 @@ function block_core_navigation_link_build_variations() {
 				$variations[] = $variation;
 			}
 		}
+
+		// If any of the post types have `has_archive` set to true then add a post-type-archive variation.
+		$has_archive = array_filter(
+			$post_types,
+			function ( $post_type ) {
```

Comment:
This closure can be declared as static (might improve performance).
```suggestion
			static function ( $post_type ) {
```